clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -779,80 +772,36 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
         const bool expect_encoding =
             TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
         if (expect_encoding) {
-            const app_pc trace_pc = reinterpret_cast<app_pc>(memref.instr.addr);
-            // Clear cache entry for new encodings.
-            if (memref.instr.encoding_is_new) {
-                shard->decode_cache_.erase(trace_pc);
+            if (shard->decode_cache_ == nullptr) {
```

Comment:
Can we init this in the shard init routine?

Reply:
Done, though this code isn't duplicated currently.